airflow/models/taskinstance.py

```diff
@@ -2069,15 +2069,27 @@ def get_rendered_template_fields(self, session: Session = NEW_SESSION) -> None:
                 setattr(self.task, field_name, rendered_value)
             self.task = task
             return
+
         try:
-            self.render_templates()
+            # Task was never executed. Initialize RenderedTaskInstanceFields
+            # to render template and mask secrets. Set MASK_SECRETS_IN_LOGS
+            # to True to enable masking similar to task run.
+            original_value = settings.MASK_SECRETS_IN_LOGS
+            settings.MASK_SECRETS_IN_LOGS = True
```

Comment:
What code does this value affect? This monkeypatching behaviour looks too hacky from a quick glance.

Reply:
```python
def mask_secret(secret: Union[str, dict, Iterable], name: Optional[str] = None) -> None:
    """
    Mask a secret from appearing in the task logs.

    If ``name`` is provided, then it will only be masked if the name matches
    one of the configured "sensitive" names.

    If ``secret`` is a dict or a iterable (excluding str) then it will be
    recursively walked and keys with sensitive names will be hidden.
    """
    # Delay import
    from airflow import settings

    # Filtering all log messages is not a free process, so we only do it when
    # running tasks
    if not settings.MASK_SECRETS_IN_LOGS or not secret:
        return

    _secrets_masker().add_mask(secret, name)
```

If that is False nothing is added to the masker.

I agree, we should change that behaviour so that instead we always add things to the mask list, and only have the filter do anything if that setting is True.

Reply:
I agree this is setting a flag that has the side_effect of masking it in template rendering which seems slightly unrelated. This setting is done in task run command and I tries to emulate this here. We can change signature to add a parameter like force but open to better approach. The suggestion to make this parameter to filter also seems good.

Reply:
Hey guys, thanks for taking the effort of fixing this. Do we know what's blocking us?

Reply:
Static check is failing.

Reply:
static check should be fixed now and rebased to latest main branch. Sorry, I didn't have docker setup for running pre-commit while starting PR.

Regarding moving the `MASK_SECRETS_IN_LOGS` check to `filter` method in `SecretsMasker` I saw some test errors on the change. This change also set and reset the variable in finally block on per request basis like task command in cli that does it per invocation.

Reply:
This one LGTM and fixes the not-nice behaviour @ashb @uranusjr  - are you ok with that one too ?